frontend/viewer/src/lib/entry-editor/field-editors/MultiOptionEditor.test.ts

```diff
@@ -0,0 +1,179 @@
+import {readable} from 'svelte/store'
+import {beforeEach, describe, expect, test} from 'vitest'
+
+import {render, screen} from '@testing-library/svelte'
+import userEvent, {type UserEvent} from '@testing-library/user-event'
+import {getState} from '../../utils/test-utils'
+import MultiOptionEditor from './MultiOptionEditor.svelte'
+
+const value = ['2', '3', '4'];
+const options = ['1', '2', '3', '4', '5'].map(id => ({id}));
+
+const context = new Map<string, unknown>([
+  ['writingSystems', readable({
+    analysis: [],
+    vernacular: [{
+      id: 'test',
+    }],
+  })],
+  ['currentView', readable({
+    fields: {'test': {show: true}},
+  })],
+]);
+
+const reusedProps = {
+  id: 'test',
+  wsType: 'vernacular',
+  name: 'test',
+  readonly: false,
+} as const;
+
+let user: UserEvent;
+let component: MultiOptionEditor<string, {id: string}>;
+
+beforeEach(() => {
+  user = userEvent.setup();
+  ({component} = render(MultiOptionEditor<string, {id: string}>, {
+    context,
+    props: {
+      ...reusedProps,
+      idValues: true,
+      value,
+      options,
+      getOptionLabel: (option) => option.id,
+    }
+  }));
+});
+
+describe('MultiOptionEditor value sorting', () => {
+  test('appends new options to the end in the order they\'re selected', async () => {
+    await user.click(screen.getByRole('textbox'));
+    await user.click(screen.getByLabelText('1'));
+    await user.click(screen.getByLabelText('5'));
+    await user.click(screen.getByRole('button', {name: 'Apply'}));
+    expect(getState(component).value).toStrictEqual(['2', '3', '4', '1', '5']);
+  });
+
+  test('removes deselected options without changing the order', async () => {
+    await user.click(screen.getByRole('textbox'));
+    await user.click(screen.getByLabelText('3'));
+    await user.click(screen.getByRole('button', {name: 'Apply'}));
+    expect(getState(component).value).toStrictEqual(['2', '4']);
+  });
+
+  test('moves double-toggled options to the end', async () => {
+    await user.click(screen.getByRole('textbox'));
+    await user.click(screen.getByLabelText('3'));
+    await user.click(screen.getByLabelText('3'));
+    await user.click(screen.getByRole('button', {name: 'Apply'}));
+    expect(getState(component).value).toStrictEqual(['2', '4', '3']);
+  });
+});
+
+describe('MultiOptionEditor displayed sorting', () => {
+  test('matches option sorting if `ordered` option is NOT set', async () => {
+    await user.click(screen.getByRole('textbox'));
+    await user.click(screen.getByLabelText('1'));
+    await user.click(screen.getByLabelText('5'));
+    await user.click(screen.getByRole('button', {name: 'Apply'}));
+    expect(getState(component).value).toStrictEqual(['2', '3', '4', '1', '5']);
+    expect(screen.getByRole<HTMLInputElement>('textbox').value).toBe('1, 2, 3, 4, 5');
+  });
+
+  test('matches values sorting if `ordered` option is set', async () => {
+    component.$set({ordered: true});
```

Comment:
Yes. That's demonstrated by the test above this one.

Reply:
so maybe should we call it `preserveOrder`? or something similar, otherwise my intuition says if ordered is true then it will be sorted for me, which seems like to reverse of what this is doing

Reply:
We could also make this the default and then using either `sortValues: boolean = false`
or `collection[Type]: 'set' | 'list' = 'list'` (potentially with no default).

But, I'll go with `preserveOrder` for now👍